Dalamud/Game/Marketboard/MarketBoard.cs

```diff
@@ -0,0 +1,154 @@
+
+    /// <inheritdoc/>
+    public event IMarketBoard.MarketBoardOfferingsReceivedDelegate? MarketBoardOfferingsReceived;
+
+    /// <inheritdoc/>
+    public event IMarketBoard.MarketBoardPurchaseRequestedDelegate? MarketBoardPurchaseRequested;
+
+    /// <inheritdoc/>
+    public event IMarketBoard.MarketTaxRatesReceivedDelegate? MarketTaxRatesReceived;
+
+    /// <inheritdoc/>
+    public void DisposeService()
+    {
+    }
+
+    private void OnMbHistory(MarketBoardHistory marketBoardHistory)
+    {
+        this.MarketBoardHistoryReceived?.Invoke(marketBoardHistory);
+    }
+
+    private void OnPurchase(MarketBoardPurchase marketBoardHistory)
+    {
+        this.MarketBoardItemPurchased?.Invoke(marketBoardHistory);
+    }
+
+    private void OnOfferings(MarketBoardCurrentOfferings currentOfferings)
+    {
+        this.MarketBoardOfferingsReceived?.Invoke(currentOfferings);
+    }
+
+    private void OnPurchaseSent(MarketBoardPurchaseHandler purchaseHandler)
+    {
+        this.MarketBoardPurchaseRequested?.Invoke(purchaseHandler);
+    }
+
+    private void OnTaxRates(MarketTaxRates taxRates)
+    {
+        this.MarketTaxRatesReceived?.Invoke(taxRates);
+    }
+}
+
+/// <summary>
+/// Plugin scoped version of MarketBoard.
+/// </summary>
+[PluginInterface]
+[InterfaceVersion("1.0")]
+[ServiceManager.ScopedService]
+#pragma warning disable SA1015
+[ResolveVia<IMarketBoard>]
+#pragma warning restore SA1015
+internal class MarketBoardPluginScoped : IInternalDisposableService, IMarketBoard
+{
+    [ServiceManager.ServiceDependency]
+    private readonly MarketBoard marketBoardService = Service<MarketBoard>.Get();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MarketBoardPluginScoped"/> class.
+    /// </summary>
+    internal MarketBoardPluginScoped()
+    {
+        this.marketBoardService.MarketBoardHistoryReceived += this.OnMarketBoardHistoryReceived;
+        this.marketBoardService.MarketBoardItemPurchased += this.OnMarketBoardItemPurchased;
+        this.marketBoardService.MarketBoardOfferingsReceived += this.OnMarketBoardOfferingsReceived;
+        this.marketBoardService.MarketBoardPurchaseRequested += this.OnMarketBoardPurchaseRequested;
+        this.marketBoardService.MarketTaxRatesReceived += this.OnMarketTaxRatesReceived;
+    }
+
+    /// <inheritdoc/>
+    public event IMarketBoard.MarketBoardHistoryReceivedDelegate? MarketBoardHistoryReceived;
+
+    /// <inheritdoc/>
+    public event IMarketBoard.MarketBoardItemPurchasedDelegate? MarketBoardItemPurchased;
+
+    /// <inheritdoc/>
+    public event IMarketBoard.MarketBoardOfferingsReceivedDelegate? MarketBoardOfferingsReceived;
+
+    /// <inheritdoc/>
+    public event IMarketBoard.MarketBoardPurchaseRequestedDelegate? MarketBoardPurchaseRequested;
+
+    /// <inheritdoc/>
+    public event IMarketBoard.MarketTaxRatesReceivedDelegate? MarketTaxRatesReceived;
+
+    /// <inheritdoc/>
+    void IInternalDisposableService.DisposeService()
```

Comment:
Need to include `this.OnMarketBoardHistoryReceived = null` for each events the plugin scoped service include, or else dangling methods will not be cleared. For an example see ChatGuiPluginScoped

Reply:
Nice catch, have fixed this in latest commit